packages/jest-cli/src/TestRunner.js

```diff
@@ -238,10 +247,22 @@ class TestRunner {
         aggregatedResults.success =
           aggregatedResults.numFailedTests === 0 &&
           aggregatedResults.numRuntimeErrorTestSuites === 0;
-        if (reporter.onRunComplete) {
-          reporter.onRunComplete(config, aggregatedResults);
-        }
-        return aggregatedResults;
+
+        return this._hasteMap.matchFiles(/\.snap/).then(res => {
+          aggregatedResults.snapshotsUnlinked = 0;
+          res.forEach(snapFile => {
+            const parts = snapFile.split('__snapshots__/');
+            const testFile = parts[0] + parts[1].slice(0, -5);
+            if (!fileExists(testFile)) {
+              fs.unlinkSync(snapFile);
+              aggregatedResults.snapshotsUnlinked++;
```

Comment:
this should only happen when the update flag is given, right? I think we should add a test failure if the update flag isn't given and remove it if `-u` is given.